modules/mobile/spec/request/v1/immunizations_request_spec.rb

```diff
@@ -514,30 +224,61 @@
       it 'returns the correct page and number of records' do
         VCR.use_cassette('lighthouse_health/get_immunizations', match_requests_on: %i[method uri]) do
           get '/mobile/v1/health/immunizations', headers: iam_headers, params: { page: { size: 2, number: 3 } }
-          ids = response.parsed_body['data'].map { |i| i['id'] }
-
-          # these are the fifth and sixth records in the vcr cassette
-          expect(ids).to eq(%w[I2-LA34JJPECU7NQFSNCRULFSVQ3M000000 I2-DOUHUYLFJLLPSJLACUDAJF5GF4000000])
         end
+
+        ids = response.parsed_body['data'].map { |i| i['id'] }
+
+        # these are the fifth and sixth from last records in the vcr cassette
+        expect(ids).to eq(%w[I2-7PQYOMZCN4FG2Z545JOOLAVCBA000000 I2-JYYSRLCG3BN646ZPICW25IEOFQ000000])
       end
 
       it 'creates navigation links for the client' do
+        size = 2
+        number = 3
+
         VCR.use_cassette('lighthouse_health/get_immunizations', match_requests_on: %i[method uri]) do
-          size = 2
-          number = 3
           get '/mobile/v1/health/immunizations', headers: iam_headers, params: { page: { size: size, number: number } }
-          expected_links = {
-            'self' => "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=#{number}",
-            'first' => "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=1",
-            'prev' =>
-              "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=#{number - 1}",
-            'next' =>
-              "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=#{number + 1}",
-            'last' => "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=8"
-          }
+        end
 
-          expect(response.parsed_body['links']).to eq(expected_links)
+        expected_links = {
+          'self' => "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=#{number}",
+          'first' => "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=1",
+          'prev' =>
+            "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=#{number - 1}",
+          'next' =>
+            "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=#{number + 1}",
+          'last' => "http://www.example.com/mobile/v1/health/immunizations?page[size]=#{size}&page[number]=8"
+        }
+
+        expect(response.parsed_body['links']).to eq(expected_links)
+      end
+    end
+
+    describe 'record order' do
```

Comment:
This tests the change in question. All other changes are to accommodate the change or have been explained in other comments.